homeassistant/components/ipma/weather.py

```diff
@@ -64,6 +66,23 @@
         _LOGGER.error("Latitude or longitude not set in Home Assistant config")
         return
 
+    station = await async_get_station(hass, latitude, longitude)
+
+    async_add_entities([IPMAWeather(station, config)], True)
```

Comment:
We should just pass in `async_setup_platform` when we have `async_setup_entry`.

Reply:
Let me just reference: https://github.com/home-assistant/architecture/issues/143

I personally think there is room for both, and should not remove the yaml configuration path.

Reply:
Sorry, I thought we should move config to component instead of platform. I don't want to remove config option. We usually set up the component and forward to platforms via config entry. That's why we don't use `async_setup_platform` usually when we have config entries. But there's nothing say about hg we have to do it like that. So I guess we're not clear about what should be the rule here.

See https://github.com/home-assistant/architecture/issues/142.

Reply:
My thinking was that almost every portuguese HA user already has the configuration.yaml with weather.ipma configured as a platform.
 
I do not want to create a breaking change, so this way new users can use the config flow, old users don't need to change a line.

Reply:
That's a valid concern. But I also like streamlining options for implementing things. It's easier if there's only one way for each feature that we want to support. I hope the architecture issue will shed some light on the way forward.